cdk/README.md

```diff
@@ -0,0 +1,130 @@
+# Welcome to ConsoleMe CDK Deployment
+
+## Motivation
+
+ConsoleMe is a web service that makes AWS IAM permissions and credential management easier for end-users and cloud administrators.
+
+Recently, I started to work on a new project and found the need to deploy ConsoleMe in a way that it would be available for multiple users, authenticating with the corporate authentication tools and scaled without managing servers.
+
+## Architecture
+
+The main idea is to use serverless services in order to remove the need from managing servers.
+
+![ConsoleMe on ECS Architecture](architecture.png "ConsoleMe on ECS Architecture")
+
+## Usage
+
+### Pre-requisites
+
+- Domain name managed with a public hosted zone on AWS Route 53.
+  Please collect this information and fill the `config.yaml` file with the hosted zone name and hosted zone id from Route 53.
+
+  ```
+  $ cp config.example.yaml config.yaml
+  ```
+
+- MacOS / Linux computer with Docker: https://docs.docker.com/get-docker/
+- NodeJS 12 or later AWS CDK command line interface installed on your computer.
+  You can easily install AWS CDK command line interface globally using `npm`:
+
+  ```
+  $ npm install -g aws-cdk
+  ```
+
+- AWS CDK assume role credential plugin. we're using this plugin in order to ease the deployment on a multi-account environment.
+  You can easily install AWS CDK assume role credential plugin globally using `npm`:
+
+  ```
+  $ npm install -g cdk-assume-role-credential-plugin
+  ```
+
+- Python 3.6 and up with Pipenv dependencies & virtual environment management framework.
+  You can easily install Pipenv command line interface it using `pip`:
+
+  ```
+  $ pip install --upgrade pipenv
+  ```
+
+- Bootstrapped CDK environment using the `modern` bootstrap template.
+  This is required to be done on each account in order to enable the `trust` functionality
+  which is required by the `cdk-assume-role-credential-plugin`.
+
+  Main account bootstrapping:
+
+  ```
+  $ cdk bootstrap --cloudformation-execution-policies arn:aws:iam::aws:policy/AdministratorAccess --plugin cdk-assume-role-credential-plugin --context bootstrap=true
```

Comment:
Move `pipenv install --dev` and `pipenv shell` instructions before the cdk command. cdk and boto3 are presumably not installed at this point.

Also instruct user to configure credentials for the AWS account they want to deploy ConsoleMe to.

Reply:
Hi, I haven't encountered any ECR dependency for that command to run, neither ran `get-login-password`. Did you encountered this issue?

Reply:
In regards to `pipenv` and `cdk`, `cdk` is installed globally via `npm install` command as described in the `Pre-requisites` stage.

Reply:
Regarding ECR, yes I did encounter that login error. It may b  because we use temporary IAM role creds and switch them out often? 

I did encounter a boto3 error when I ran the cdk command before creating my virtualenv:
![image](https://user-images.githubusercontent.com/7538233/123578875-9b3fae80-d78b-11eb-8fe7-f2919211694d.png)
p

Another question: How do I tell the `cdk-assume-role-credential-plugin` which role name to assume in my environment?

Thank you.

Reply:
Yes, you do need to run `pipenv install --dev` & `pipenv shell` before running the `cdk bootstrap` command if you don't have `boto3` installed globally with `pip` so that's a good point to instruct running it.

Reply:
Here is how the deployment looks like in my environment (I did not run ECR login or something, just assumed a role with Administrator access):
1. I ran `cdk deploy --all` after creating the `config.yaml` file and altering the `hosted_zone_id`, `hosted_zone_name` and `spoke_accounts` configuration variables.
2. Docker packaging of the configuration lambda:
![Screen Shot 2021-06-28 at 9 52 01](https://user-images.githubusercontent.com/827024/123593068-c14e6880-d7f6-11eb-8552-c73a4720085b.png)
3. Deployment immediately after packaging is done:
![Screen Shot 2021-06-28 at 9 52 15](https://user-images.githubusercontent.com/827024/123593139-d9be8300-d7f6-11eb-875d-b06c1f54dd70.png)

Reply:
@castrapel Let me know if you have additional concerns or we can proceed.